spec/index.bs

```diff
@@ -3878,6 +3853,56 @@ Issue: link to the semantics of SV_SampleIndex and SV_Coverage in WGSL spec.
         </div>
 </dl>
 
+<div algorithm>
+    <dfn abstract-op>validating GPURenderPipelineDescriptor</dfn>(descriptor, features)
+        **Arguments:**
+            - {{GPURenderPipelineDescriptor}} |descriptor|
+            - [=list=]&lt;{{GPUFeatureName}}&gt; |features|
+
+        Return `true` if all of the following conditions are satisfied:
+
+            - [$validating GPUProgrammableStageDescriptor$]({{GPUShaderStage/VERTEX}},
+                |descriptor|.{{GPURenderPipelineDescriptor/vertexStage}},
+                |descriptor|.{{GPUPipelineDescriptorBase/layout}}) succeeds.
+            - If |descriptor|.{{GPURenderPipelineDescriptor/fragmentStage}} is not `null`:
+                - [$validating GPUProgrammableStageDescriptor$]({{GPUShaderStage/FRAGMENT}},
+                    |descriptor|.{{GPURenderPipelineDescriptor/fragmentStage}},
+                    |descriptor|.{{GPUPipelineDescriptorBase/layout}}) succeeds.
+            - If |descriptor|.{{GPURenderPipelineDescriptor/fragmentStage}} is `null`:
+                - |descriptor|.{{GPURenderPipelineDescriptor/colorStates}} is empty.
+            - |descriptor|.{{GPURenderPipelineDescriptor/colorStates}}.length is less than
+                or equal to 4.
```

Comment:
nit: GPULimits.maxColorAttachments (or we can add an `Issue` to define that limit)